core/src/main/scala/org/apache/spark/deploy/client/TestClient.scala

```diff
@@ -49,7 +49,8 @@ private[spark] object TestClient {
     val conf = new SparkConf
     val rpcEnv = RpcEnv.create("spark", Utils.localHostName(), 0, conf, new SecurityManager(conf))
     val desc = new ApplicationDescription("TestClient", Some(1), 512,
-      Command("spark.deploy.client.TestExecutor", Seq(), Map(), Seq(), Seq(), Seq()), "ignored")
+      Command("org.apache.spark.deploy.client.TestExecutor", Seq(), Map(), Seq(), Seq(), Seq()),
```

Comment:
I agree that would be better.  Since `TestExecutor` is an object, it looks like I would have to use `TestExecutor.getClass.getCanonicalName` which returns "org.apache.spark.deploy.client.TestExecutor$" and then the $ would need to be removed.  

So it's slightly more complicated, but still probably more future proof.  Let me implement that and verify everything still works.